### DISCUSSION TITLE
Display sensor values

### DISCUSSION THREAD

**Discussion by ivanvach** [arendst/Tasmota#21123] [Q&A]
I do have Tasmota running on Shelly 1 ( ESP8266 w/ 2MB  ) plus I have two I2C devices connected

BME280 I2C temperature/humidity/pressure sensor 
SH1106 OLED display 1.3" connected over I2C as well

Both do work separately and independently. 

Display does work with Tasmota binary  ....display.bin
Sensor works with Tasmota binary    ...sensors.bin

How to make them work together and the display to show the sensor values - temperature, humidity, dew point, atmospheric pressure?

Thanks

**Comment by barbudor** [comment]
You have to build your own Tasmota variant that combines features-set from both variants
Read https://tasmota.github.io/docs/Compile-your-build/

**Comment by ivanvach** [comment]
Thanks. Done that and now I do have indeed both display and sensor working. The question now is how to get a decent display.
I use this Tasmota command

DisplayMode 3

, and it does display the readings of the sensor in a very unattractive and inconvenient fashion. It shows two lines of information only in a very small font and scrolls between the lines without a steady screen. Furthermore it uses about 10% of the viewable area of my 1.3" display.

Ideally I would like to see 5 lines

1. Day, date and time
2. Temperature (the larges possible size to fill the screen and allow for 3 lines below)
3. Humidity
4. Pressure

The only way for me to do my own design would be with the command DisplayText, however how do I display the values from the sensor.

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Displays/#display-json-variables should work for sensors.
https://tasmota.github.io/docs/Displays/#batch-files allows a screen design in a file.
(With 2 MB of flash, you can have a build including a file system)

**Comment by ivanvach** [comment]
How to display the day of the week ?

Almost there. Using Rule1 command (shared below) I managed to almost achieve my objective. The only thing remaining is how to display the day of the week. Anyone who has SH1106 OLED display 1.3" and BME280 sensor feel free to use my Rule1. It does work.

Rule1 on tele-BME280#Temperature do DisplayText[zf1s1t] [x70T] [x0y10h144] [x0y20s3t] [s3x0y15] %value% endon       on tele-BME280#Humidity do DisplayText [x0y50h144] [x0y55s1] %value% endon       on tele-BME280#Pressure do DisplayText [x50y55s1] %value% hPa endon

**Reply by sfromis** [reply]
AFAIK, no direct access to day of week, but if you set up 7 timers, one for each day of the week at 00:00, you can use the associated trigger to set a variable to the number, or a trigger for each day to set a variable to the text.

BTW.... What I posted was about doing the sensor display without using rules or even directly `DisplayText` commands, as you can have JSON-based values updated, and at shorter refresh interval than telemetry period.

**Reply by ivanvach** [reply]
Not sure that I understand the idea of 7 timers well enough but I have the feeling that it will be affected by power outages. How about day of the week to be delivered as MQTT message by the home automation computer like that
{"weekday":"Tue"}
The question is how to do that? Do I use topic cmnd/tasmota-xxx/ and if yes than what command or something else. Pls advise.

**Reply by sfromis** [reply]
To set a variable from your backend, a simple approach could be to do something like publishing to topic `cmnd/tasmota-xxx/var1` and payload the value to set in the variable. 

When it comes to using timers, one timer per weekday would be the number (trigger `Clock#Timer=3` etc), and then you could set the matching weekday in a `mem` variable; these are persistent across boots. But yes, such a timer would get lost if lacking power at midnight (or the selected time).

**Reply by ivanvach** [reply]
Thanks. How would I display the value of var1 per your example above. Is it subject to DisplayText command or something else.

**Reply by sfromis** [reply]
Using `%var1%` will be expanded before the rule starts, no matter what commands you use in the rule, in the same way as `%value%` you already use.

**Reply by ivanvach** [reply]
Thanks. Works like a charm.

**Comment by ivanvach** [comment]
Here is an image of the display. Mission accomplished !

https://www.amazon.com/photos/shared/FZcYK8XVRAmx8GMY9dmgXw.jNina4xf5cJ5FhGdWh3U8Y
